src/Castle.Core/DynamicProxy/MixinData.cs

```diff
@@ -38,28 +44,77 @@ public MixinData(IEnumerable<object> mixinInstances)
 			{
 				var sortedMixedInterfaceTypes = new List<Type>();
 				var interface2Mixin = new Dictionary<Type, object>();
+				delegateMixinCount = 0;
 
 				foreach (var mixin in mixinInstances)
 				{
-					var mixinInterfaces = mixin.GetType().GetInterfaces();
+					Type[] mixinInterfaces;
+					object target;
+					if (mixin is MulticastDelegate @delegate)
```

Comment:
The `@delegate` variable isn't used.

Since all delegates are `MulticastDelegate`s and therefore also `Delegate`s either type could be used here, but it seems a tad strange to see `MulticastDelegate` used here when it isn't needed. I understand the use in `IsDelegateType`.

https://stackoverflow.com/questions/4833053/why-delegate-types-are-derived-from-multicastdelegate-class-why-not-it-directly

Reply:
I remembered you mentioning your preference for `MulticastDelegate` at an earlier time and went with that... probably a little too far in this case.

I changed the check from `is MulticastDelegate @delegate` to just `is Delegate`.

Reply:
Yer I recall that too but don't remember why, and I saw I had a link to an article on the MSDN blogs about why. That page has since been killed with the recent change to remove MSDN branding from domains and I couldn't find the article.

Today 😉 I think we should use `Delegate` wherever possible, since `MulticastDelegate` should have been merged into `Delegate` nearly 20 years ago.

Reply:
Agreed. So i suppose `MulticastDelegate` is only useful to *quickly* check whether a type is a delegate type, as `type.IsSubclassOf(typeof(MulticastDelegate))` does less work than `typeof(Delegate).IsAssignableFrom(type)`.